### PR TITLE
IRGen: Emit a weak reference to swift_async_extendedFramePointerFlags when neccessary

### PR DESCRIPTION
**Pull request by aschwaighofer**
When we deploy to a minimum target that is not known to support extended
frame information the function prolog of Swift async functions will
contain a reference to swift_async_extendedFramePointerFlags. This
reference needs to be weak like any async symbols.

rdar://83412550

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please test

File: lib/IRGen/IRGenModule.cpp [hunk 1]
```diff
@@ -1776,3 +1777,45 @@ TypeExpansionContext IRGenModule::getMaximalTypeExpansionContext() const {
 const TypeLayoutEntry &IRGenModule::getTypeLayoutEntry(SILType T) {
   return Types.getTypeLayoutEntry(T);
 }
+
+/// Return whether FrameLowering should always set the "extended frame
+/// present" bit in FP, or set it based on a symbol in the runtime.
+static bool isSwiftAsyncContextIsDynamicallySet(llvm::Triple TT) {
```
Comment (DougGregor) [inline lib/IRGen/IRGenModule.cpp:1783]: Can you use `isConcurrencyAvailable` from `GenMetadata.cpp` rather than hard-coding these values again?

**Review by DougGregor — APPROVED** [review]
Looks good, with one request to re-use the existing check for "is concurrency guaranteed to be there?"

**Comment by aschwaighofer** [comment]
@swift-ci test

**Pull request merged**